File: L16_Delete_Middle_Node.java
class Node {
    public int data;
    public Node next;

    Node()
    {
        this.data = 0;
        this.next = null;
    }

    Node(int data)
    {
        this.data = data;
        this.next = null;
    }

    Node(int data, Node next)
    {
        this.data = data;
        this.next = next;
    }
};

public class L16_Delete_Middle_Node {
    public static Node deleteMiddle(Node head) {
        // Write your code here.
        if(head.next == null) return null;
        Node slow = head;
        Node fast = head;
        Node prevNode = null;
        
        while(fast != null && fast.next != null){
            prevNode = slow;
            slow = slow.next;
            /*if(fast.next.next != null)*/fast = fast.next.next;
            //else return slow;
        }
        prevNode.next = slow.next;
        slow.next = null;
        return head;
    }
}

File: L16_Delete_Middle_Node.cpp

#include<iostream>
#include<bits/stdc++.h>
using namespace std;

class Node{
public:
    int data;
    Node *next;
    Node()
    {
        this->data = 0;
        next = NULL;
    }
    Node(int data)
    {
        this->data = data;
        this->next = NULL;
    }
    Node(int data, Node* next)
    {
        this->data = data;
        this->next = next;
    }
};

bool detectCycle(Node *head){
	//	Write your code here
    if(head->next == nullptr) return false;
        Node *slow = head;
        Node *fast = head;

        while(fast->next != nullptr && fast->next->next != nullptr){
            slow = slow->next;
            fast = fast->next->next;
            if(slow == fast) return true;
        }
        return false;
}